src/main/scala/edu/scalanus/interpreter/ScalanusInterpreter.scala

```diff
@@ -0,0 +1,147 @@
+package edu.scalanus.interpreter
+
+import javax.script.ScriptContext
+
+import edu.scalanus.ir._
+
+object ScalanusInterpreter {
+
+  def eval(ir: IrNode, context: ScriptContext): AnyRef = {
+    context match {
+      case scalanusContext: ScalanusScriptContext => eval(ir, scalanusContext,scalanusContext.PROGRAM_SCOPE)
+        .asInstanceOf[AnyRef]
+      case _ => throw new IllegalArgumentException("Not Scalanus Context") // TODO: exception
+    }
+  }
+
+  def eval(ir: IrNode, context: ScalanusScriptContext, scope:Int): Any = {
+    ir match {
+      case irProgram:IrProgram => evalProgram(irProgram,context,scope)
+      case irBlock:IrBlock => evalBlock(irBlock,context,scope)
+      case irRef:IrRef => evalRef(irRef,context,scope)
+      case irStmt:IrStmt => evalStmt(irStmt,context,scope)
+      case irPattern:IrPattern => evalPattern(irPattern,context,scope)
+      case irExpr:IrExpr => evalExpr(irExpr,context,scope)
+      case _ => ???
```

Comment:
pomięcie tego case będzie równoważne, tylko dostaniemy MatchError zamiast NotImplementedError ale to nie powinno się wydarzyć